python/paddle/fluid/tests/unittests/test_lookup_table_bf16_op.py

```diff
@@ -0,0 +1,193 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+from paddle.fluid.tests.unittests.op_test import (
+    OpTest, convert_float_to_uint16, convert_uint16_to_float,
+    skip_check_grad_ci)
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+from paddle.fluid.op import Operator
+from paddle import enable_static
+
+
+def _lookup(weights, ids):
+    w_shape = weights.shape
+    out_shape = list(ids.shape[:-1])
+    out_shape.append(w_shape[-1])
+    out = weights[ids.flatten()].reshape(out_shape)
+    return out
+
+
+def _get_grad(weights, ids):
```

Comment:
Please take into consideration moving _get_grad and lookup functions and common parts of TestLookupTableBF16Op and TestLookupTableBF16OpIds4D to a new parent Test class  just like you did nicely with TestLookupTableBF16OpWIsSelectedRows.

Reply:
Did some refactoring to more reuse code, however the two functions `_lookup` and `_get_grad` are more handy as standalone functions.